File: src/models/VehicleDocument.js
const mongoose = require('mongoose');

const vehicleDocumentSchema = new mongoose.Schema({
  vehicleId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Vehicle',
    required: [true, 'Le véhicule est requis'],
    index: true
  },
  ownerUserId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: [true, 'Le propriétaire est requis'],
    index: true
  },
  type: {
    type: String,
    required: [true, 'Le type de document est requis'],
    enum: {
      values: ['ASSURANCE', 'CT', 'FACTURE', 'AUTRE'],
      message: 'Le type doit être ASSURANCE, CT, FACTURE ou AUTRE'
    },
    index: true
  },
  label: {
    type: String,
    required: [true, 'Le libellé est requis'],
    trim: true,
    maxlength: [200, 'Le libellé ne peut pas dépasser 200 caractères']
  },
  fileUrl: {
    type: String,
    required: [true, 'L\'URL du fichier est requise'],
    trim: true,
    maxlength: [500, 'L\'URL ne peut pas dépasser 500 caractères']
  },
  date: {
    type: Date,
    required: [true, 'La date est requise'],
    default: Date.now
  },
  notes: {
    type: String,
    trim: true,
    maxlength: [500, 'Les notes ne peuvent pas dépasser 500 caractères']
  }
}, {
  timestamps: true
});

vehicleDocumentSchema.index({ vehicleId: 1, type: 1 });
vehicleDocumentSchema.index({ ownerUserId: 1 });
vehicleDocumentSchema.index({ vehicleId: 1, date: -1 });

module.exports = mongoose.model('VehicleDocument', vehicleDocumentSchema);

File: flutter_app/lib/models/vehicle_document.dart
class VehicleDocument {
  final String id;
  final String vehicleId;
  final String type; // ASSURANCE, CT, FACTURE, AUTRE
  final String label;
  final String fileUrl;
  final DateTime date;
  final String? notes;
  final DateTime createdAt;
  final DateTime updatedAt;

  VehicleDocument({
    required this.id,
    required this.vehicleId,
    required this.type,
    required this.label,
    required this.fileUrl,
    required this.date,
    this.notes,
    required this.createdAt,
    required this.updatedAt,
  });

  factory VehicleDocument.fromJson(Map<String, dynamic> json) {
    return VehicleDocument(
      id: json['_id'] ?? json['id'] ?? '',
      vehicleId: json['vehicleId'] ?? json['vehicle'] ?? '',
      type: json['type'] ?? '',
      label: json['label'] ?? '',
      fileUrl: json['fileUrl'] ?? '',
      date: json['date'] != null 
          ? DateTime.parse(json['date']) 
          : DateTime.now(),
      notes: json['notes'],
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      updatedAt: json['updatedAt'] != null 
          ? DateTime.parse(json['updatedAt']) 
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'type': type,
      'label': label,
      'fileUrl': fileUrl,
      'date': date.toIso8601String(),
      if (notes != null && notes!.isNotEmpty) 'notes': notes,
    };
  }

  String get typeLabel {
    switch (type) {
      case 'ASSURANCE':
        return 'Assurance';
      case 'CT':
        return 'Contrôle technique';
      case 'FACTURE':
        return 'Facture';
      case 'AUTRE':
        return 'Autre';
      default:
        return type;
    }
  }
}





